Makefile

```diff
@@ -78,8 +78,14 @@ llvm-build: llvm-build/build.ninja
 
 # Build the Go compiler.
 build/tinygo:
-	@if [ ! -f llvm-build/bin/llvm-config ]; then echo "Fetch and build LLVM first by running:\n  make llvm-source\n  make llvm-build"; exit 1; fi
+	@if [ ! -f $(LLVM_BUILDDIR)/bin/llvm-config ]; then echo "Fetch and build LLVM first by running:\n  make llvm-source\n  make llvm-build"; exit 1; fi
+	# Generate src/runtime/internal/sys/zversion.go with TinyGo version number, for GoLand and IDEA
+	echo 'package sys' > src/runtime/internal/sys/zversion.go
+	echo >> src/runtime/internal/sys/zversion.go
+	echo "const TheVersion = \`go`grep '^const version' version.go|sed 's/const version = "//'|sed 's/"//'`\`" >> src/runtime/internal/sys/zversion.go
```

Comment:
Hmm, I'm used to building TinyGo in-place.  Haven't thought about doing out-of-tree builds with it.

If out-of-tree builds are done by people, this will probably need adjusting to ensure it's using the correct base path.

Reply:
I wasn't very clear. All build results are stored in the build directory, making it easy to remove build files (exception: src/device/*/*.go generated by gen-device-svd.py). What I mean here is that I think it's better to create the zversion.go in build/tinygo/src/runtime... instead of src/runtime/...

Reply:
Ahhh.  That location of `zversion.go` is just reverse engineered from what GoLand / IntelliJ IDEA check.  Doesn't seem to have any flexibility about it. :wink:

Reply:
The workflow this fit into, is where a person wants to add TinyGo to the list of Go versions in GoLand or IntelliJ IDEA.

GoLand pops open a dialog for the user to navigate to the root of a new Go install.  When the user selects said directory:

* GoLand checks for the presence of `src/runtime/internal/sys/zversion.go`
* Makes sure it contains a readable version number (thus the formatting above)
* Checks for the presence of `bin/go` as well